File: proto.rb
#!/usr/bin/env ruby
# coding: utf-8

require_relative 'lib/unmo'

def prompt(unmo)
  "#{unmo.name}:#{unmo.responder_name}> "
end

def select_random(ary)
  ary[rand(ary.size)]
end

def save(ai, log)
  ai.save
  File.open('log.txt', 'a') do |f|
  f.puts(log)
  f.puts
  end
end

puts ('Unmo System prototype : proto')
proto = Unmo.new('proto')
log = ["Unmo System : #{proto.name} Log -- #{Time.now}"]

while true
  print '> '
  input = readline.chomp
  break if input.empty?

  response = proto.dialogue(input)
  puts prompt(proto) + response
end

save(proto, log)

File: python/proto/proto.py
class Responder:
    def __init__(self, name):
        self._name = name

    def response(self, line):
        return '{}ってなに？'.format(line)

    @property
    def name(self):
        return self._name

class Unmo:
    def __init__(self, name):
        self._name = name
        self._responder = Responder('What')

    def dialogue(self, line):
        return self._responder.response(line)

    @property
    def responder_name(self):
        return self._responder.name

    @property
    def name(self):
        return self._name

def prompt(unmo):
    return '{}:{}> '.format(unmo.name, unmo.responder_name)

def main():
    print('Unmo System prototype : proto')
    proto = Unmo('proto')
    while True:
        line = input('> ').strip()
        if not line:
            break

        response = proto.dialogue(line)
        print('{}{}'.format(prompt(proto), response))


if __name__ == '__main__':
    main()
